x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/ProjectAwayColumns.java

```diff
@@ -49,6 +50,17 @@ public PhysicalPlan apply(PhysicalPlan plan) {
                 return currentPlanNode;
             }
 
+            // for mv_expand, the target attribute should be retained in its original position
+            if (currentPlanNode instanceof MvExpandExec mvExpand) {
```

Comment:
I don't think we should special case for MvExpand; I believe there is a general bug that can re-occur whenever a plan node adds a new attribute somewhere else than on the right hand side, resp. when the new attributes don't come after all previously existing attributes in the `.output()` list.

In fact, it's a nice coincidence that this rule worked for so long without running into issues, and it probably only worked because most of our plans do append new attributes to the right.